viewing_party/party.py

```diff
@@ -1,23 +1,360 @@
+            for movie in user_data[category]:
+                # and if we find our title,
+                if title == movie["title"]:
+                    # we add the whole movie to our result list
+                    result.append(movie)
+        else: #if user is false, we're working with friends info
+            #for each friend in the list
+            for friend in user_data["friends"]:
+                #for each movie in "watched"
+                for movie in friend["watched"]:
+                    # and if we find our title and movie is not yet there from another friend,
+                    if title == movie["title"] and movie not in result:
+                        # we add the whole movie to our result list
+                        result.append(movie)
+
+    return result
+
+def get_unique_watched(user_data):
+    '''
+    - take one parameter: `user_data` 
+    - Determine which movies the user has watched, but none of their friends have watched.
+    - Return a list of dictionaries, that represents a list of movies
+    '''
+
+    # using helper function to create a list of titles of movies that user has watched
+    user_watched_list = get_user_movies_list(user_data, "watched")
+
+    # using helper function to create a list of titles of movies that friends have watched
+    all_friends_movies_list = get_friends_watched_movies_list(user_data)
+
+    #creating a list of unique titles of movies that user has watched
+    user_unique_movies_list = use_set_to_compare_lists_return_list(user_watched_list, all_friends_movies_list)
+
+    #we need this user parameter for the helper function convert_titles_into_list_of_movies_dict
+    # to tell it whether we access user data of friends data
+    user = True
+
+    final_result = convert_titles_into_list_of_movies_dict(user_data, user_unique_movies_list, user, "watched")
+
+    return final_result
+
+def get_friends_unique_watched(user_data):
+    '''
+    - take one parameter: `user_data` 
+    - Determine which movies from the friends list the user hasn't watched
+    - Return a list of dictionaries, that represents a list of movies
+    '''
+    # using helper function to create a list of titles of movies that user has watched
+    user_watched_list = get_user_movies_list(user_data, "watched")
+
+    # using helper function to create a list of titles of movies that friends have watched
+    all_friends_movies_list = get_friends_watched_movies_list(user_data)
+
+    #creating a list of unique titles of friends movies that user hasn't watched
+    friends_unique_movies_list = use_set_to_compare_lists_return_list(all_friends_movies_list, user_watched_list)
+
+    #we need this user parameter for the helper function convert_titles_into_list_of_movies_dict
+    # to tell it that we want to access friends data
+    user = False
+
+    final_result = convert_titles_into_list_of_movies_dict(user_data, friends_unique_movies_list, user, "")

```

Comment:
Since user is a parameter, you could absolutely just put True or False in its place instead of making it its own variable on line 263.